pkg/interface/chat/src/js/components/new.js

```diff
@@ -51,91 +57,90 @@ export class NewScreen extends Component {
       return;
     }
 
-    let station = `~${window.ship}/${state.idName}`;
-    let actions = [
-      {
-        create: {
-          nom: state.idName,
-          des: "chatroom",
-          sec: "village"
-        }
-      },
-      {
-        source: {
-          nom: 'inbox',
-          sub: true,
-          srs: [station]
-        }
-      }
-    ];
-
+    let station = `/${state.idName}`;
 
-    if (state.invites.length > 0) {
+    if (station in props.inbox) {
+      this.setState({
+        inviteError: false,
+        idError: true,
+        success: false
+      });
+      return;
+    }
 
-      let aud = state.invites.split(',')
+    // TODO: send invites
+    let aud = [];
+    let isValid = true;
+    if (state.invites.length > 2) {
+      aud = state.invites.split(',')
         .map((mem) => mem.trim())
-        .map(deSig);
+        .map(deSig)
+        .map((mem) => {
+          return `~${mem}`;
+        });
 
-      let isValid = true;
       aud.forEach((mem) => {
-        if (!urbitOb.isValidPatp(`~${mem}`)) {
+        if (!urbitOb.isValidPatp(mem)) {
           isValid = false;
         }
       });
+    }
 
-      if (isValid) {
-        actions.push({
-          permit: {
-            nom: state.idName,
-            sis: aud,
-            inv: true
-          }
-        });
-
-        actions.push({
-          phrase: {
-            aud: aud.map((aud) => `~${aud}/i`),
-            ses: [{
-              inv: {
-                inv: true,
-                cir: station
-              }
-            }]
-          }
-        });
+    if (!isValid) {
+      this.setState({
+        inviteError: true,
+        idError: false,
+        success: false
+      });
+      return;
+    }
 
-        if (this.textarea) {
-          this.textarea.value = '';
-        }
+    if (this.textarea) {
+      this.textarea.value = '';
+    }
 
-        this.setState({
-          inviteError: false,
-          idError: false,
-          success: true,
-          invites: ''
-        }, () => {
-          props.setSpinner(true);
-          props.api.chat(actions);
-        });
+    // TODO: don't do this, it's shitty
```

Comment:
Should it not be done, then? 😄

Reply:
Lol will look into this

Reply:
After looking at it, I think I should continue doing the shitty thing until we get designers to look at it and come up with a better UX around fine-grained permission models being generated.

For context, what it's doing is...

Parsing which security type the chatroom will be, then taking the ships in the invite box, inviting them and granting them permission to the channel.

So you can't create a channel that already has banned users. This seems fine to me and retains the semantic around how the New chatroom page has functioned, but it's not taking full advantage of the security features. Seems fine, if a bit ugly in the code.